tests/network/sriov.go

```diff
@@ -697,6 +715,34 @@ func waitVMI(vmi *v1.VirtualMachineInstance) (*v1.VirtualMachineInstance, error)
 	return virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
 }
 
+// deleteVMI deletes the specified VMI and waits for its absence.
+// Waiting for the VMI removal is placed intentionally for VMI/s with SR-IOV networks in order
+// to assure resources (VF/s) are fully released before reused again on a new VMI.
+func deleteVMI(vmi *v1.VirtualMachineInstance) error {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		panic(err)
+	}
+
+	GinkgoWriter.Println("sriov:", vmi.Name, "deletion started")
+	if err = virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &k8smetav1.DeleteOptions{}); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to delete VMI: %v", err)
+	}
+
+	const timeout = 30 * time.Second
+	err = wait.Poll(1*time.Second, timeout, func() (done bool, err error) {
+		_, getErr := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
+		if k8serrors.IsNotFound(getErr) {
+			return true, nil
+		}
+		return false, getErr
+	})
+	return fmt.Errorf("failed to wait for the VMI to disapear: %v", err)
```

Comment:
nit: I'd find this easier to read if it were simply:
```golang
return wait.Poll(1*time.Second, timeout, func() (done bool, err error) {
		_, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &k8smetav1.GetOptions{})
		if k8serrors.IsNotFound(err) {
			return true, nil
		}
		return false, err
	})
```

Reply:
I wanted to differentiate between the waiting timeout error and the error of the deleting action.
Do you think it is not needed?

Reply:
Doesn't `wait.Poll` [return an error](https://github.com/kubernetes/apimachinery/blob/97e5df2d0258ac077093867fabf508c1fc31f53b/pkg/util/wait/wait.go#L592) if it times out ? 

If so, the thrown error will just be different (timeout vs error in the caller), won't it ?

Reply:
I think you are right then... lets do this.

Reply:
Please consider using `PollImmediate`, it doesn't wait the interval before checking the condition in the first time from what i understand.

Reply:
Done

Reply:
> While changing, please consider using `PollImmediate`, it doesn't wait the interval before checking the condition in the first time from what i understand.

Fine by me. Done